cmd/storagenode-updater/main.go

```diff
@@ -136,62 +154,76 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 	return nil
 }
 
-func update(ctx context.Context, nodeID storj.NodeID) (err error) {
-	client := checker.New(runCfg.ClientConfig)
+func cmdRecover(cmd *cobra.Command, args []string) (err error) {
+	err, closeLog := openLog()
+	defer func() { err = errs.Combine(err, closeLog()) }()
 
-	currentVersion, err := binaryVersion(runCfg.BinaryLocation)
+	log.Println("storagenode updater recovering!")
+
+	badExec := strings.Replace(os.Args[0], ".backup", "", 1)
+	log.Printf("deleting bad updater binary: %s\n", badExec)
+	err = os.Remove(badExec)
+	if err != nil && !os.IsNotExist(err) {
+		log.Println("")
+	}
+
+	log.Printf("restoring backup binary from: %s\n", os.Args[0])
+	if err := os.Rename(os.Args[0], badExec); err != nil {
```

Comment:
If the recovery is run from a binary with non `.backup` in the filename this is going to fail because the binary is removed in the line 165.

I think that we should return an error if recovery isn't called from a `.backup` binary, but I'm OK with another strategy that doesn't fail here for the reason commented above.